File: Frontend/SeatifyFrontend/src/app/models/event-occurrence.ts
export interface EventOccurrenceEvent {
  id: string;
  name: string;
  description: string;
  primaryColor?: string;
  secondaryColor?: string;
  logoImageUrl?: string;
  currency?: string;
}

export interface EventOccurrenceVenue {
  id: string;
  name: string;
}

export interface EventOccurrenceAuditorium {
  id: string;
  name: string;
  currency?: string;
}

export interface EventOccurrence {
  id: string;
  eventId: string;
  venueId: string;
  auditoriumId: string;
  startsAtUtc: string;
  endsAtUtc: string;
  bookingOpenAtUtc: string;
  bookingCloseAtUtc: string;
  doorsOpenAtUtc?: string | null;
  currencyOverride?: string;
  status: string;
  effectiveCurrency: string;
  event?: EventOccurrenceEvent | null;
  venue?: EventOccurrenceVenue | null;
  auditorium?: EventOccurrenceAuditorium | null;
}

File: Backend/SeatifyBackend/Entities/Models/EventOccurrence.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Models
{
    public class EventOccurrence
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string EventId { get; set; } = string.Empty;
        public string VenueId { get; set; } = string.Empty;
        public string AuditoriumId { get; set; } = string.Empty;

        public DateTime StartsAtUtc { get; set; }
        public DateTime EndsAtUtc { get; set; }
        public string Status { get; set; } = "Published";

        public DateTime BookingOpenAtUtc { get; set; }
        public DateTime BookingCloseAtUtc { get; set; }

        public DateTime? DoorsOpenAtUtc { get; set; }

        public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;

        public virtual Event Event { get; set; } = null!;
        public virtual Venue Venue { get; set; } = null!;
        public virtual Auditorium Auditorium { get; set; } = null!;
        public virtual List<Reservation> Reservations { get; set; } = new();

        // TODO: BookingSession

    }
}
